### PR TITLE
Adding check that dbnode is up for fullscan

### PR DESCRIPTION
**Pull request by aleksbykov**
If node is down by nemesis, fullscan could though exception
that node is not reachable. Check that node is up.

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2367,7 +2367,7 @@ def run_stress_thread(self, stress_cmd, timeout, output_dir, stress_num=1, keysp
 
         if keyspace_name:
             stress_cmd = stress_cmd.replace(" -schema ", " -schema keyspace={} ".format(keyspace_name))
-        elif 'keyspace=' not in stress_cmd: # if keyspace is defined in the command respect that
+        else:
```
Comment (amoskong) [inline sdcm/cluster.py]: We won't assign keyspace in original stress cmdline, your change is fine.

Can we safely add an assert in else block?

```
        if keyspace_name:
            stress_cmd = stress_cmd.replace(" -schema ", " -schema keyspace={} ".format(keyspace_name))
        else:
            assert('keyspace=' not in stress_cmd)
            .....
```

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2852,6 +2852,10 @@ def run_in_thread():
             while current - start < duration:
                 loader_node = random.choice(self.nodes)
                 db_node = random.choice(db_nodes)
+                if db_node.remoter and not db_node.remoter.is_up():
+                    self.log.warning("Chosen node is down. Skip Request")
+                    time.sleep(interval)
+                    continue
                 self.log.info('Fullscan start on node {} from loader {}'.format(db_node.name, loader_node.name))
```
Comment (amoskong) [inline sdcm/cluster.py]: Can we add above new code after termination_event checking? it might delay the look to break when termination_event is already set.

For example, In cleanup stage, nodes are destroyed, and termination_event is set, but the new code will bright a redundant sleep.

File: sdcm/cluster.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 from avocado.utils import script
 from datetime import datetime
 from .log import SDCMAdapter
-from .remote import Remote
+from .remote import Remote, RemoteFabric
```
Comment (amoskong) [inline sdcm/cluster.py]: This doesn't related with PR subject & patch subject.
It's fine to split the change into another patch in this PR.

RemoteFabric PR  #814 is draft status, you should mention that this PR replied on  #814.
We can't apply this PR before  #814.

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py]: This change was intentionally done by me, to enable putting the keyspace in the stress commands.
I believe it was removed by mistake by alex.

File: sdcm/cluster.py (hunk 3 above)
Comment (fruch) [reply sdcm/cluster.py]: I agree we should keep those two PRs separated.

File: sdcm/cluster.py (hunk 2 above)
Comment (bentsi) [reply sdcm/cluster.py]: good catch Amos :+1:

File: sdcm/cluster.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: Not sure was done by me. may some how leaked from pr 814. Restore

File: sdcm/cluster.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: Thanks. Fixed

File: sdcm/cluster.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: Some how it is leaked from  PR #814, it is strange. Fixed.

**Comment by aleksbykov** [comment]
It is not clear, how some code from pr #814 leaked hear. i create new branch from master.

**Comment by roydahan** [comment]
@bentsi can we merge it?

**Comment by bentsi** [comment]
@aleksbykov please resolve conflicts

**Comment by aleksbykov** [comment]
Resoled.

File: sdcm/cluster.py [hunk 4]
```diff
@@ -2864,10 +2864,15 @@ def run_in_thread():
             start = current = time.time()
             while current - start < duration:
                 loader_node = random.choice(self.nodes)
-                db_node = random.choice(db_nodes)
-                self.log.info('Fullscan start on node {} from loader {}'.format(db_node.name, loader_node.name))
                 if loader_node.termination_event.isSet():
                     break
+                db_node = random.choice(db_nodes)
+                if db_node.remoter and not db_node.remoter.is_up():
+                    self.log.warning("Chosen node is down. Skip Request")
```
Comment (roydahan) [inline sdcm/cluster.py]: why not choose another one?

Comment (aleksbykov) [reply sdcm/cluster.py]: Fix. remove the sleep for timeout, which allow to restart and choose the db node again.
